components/cards/ClaimAction.tsx

```diff
@@ -13,78 +15,78 @@ export type ClaimActionProps = SidebarProps & {
+  const {
+    displayNewForSalePrice,
+    // displayNetworkFeePayment
+  } = actionData;
 
   // React.useEffect(() => {
-  //   let _transactionSubtotal = calculateWeiSubtotalField(
+  //   const _transactionSubtotal = calculateWeiSubtotalField(
   //     displayNetworkFeePayment
   //   );
 
   //   updateActionData({ transactionSubtotal: _transactionSubtotal });
   // }, [displayNetworkFeePayment]);
 
   async function _claim() {
-    // FIXME: Update for streaming payments
-    // updateActionData({ isActing: true });
-    // let baseCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase1Coord.x,
-    //   claimBase1Coord.y
-    // );
-    // let destCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase2Coord.x,
-    //   claimBase2Coord.y
-    // );
-    // let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
-    //   (v) => {
-    //     return BigNumber.from(v.toString(10));
-    //   }
-    // );
-    // if (path.length == 0) {
-    //   path = [BigNumber.from(0)];
-    // }
-    // const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
-    // const newContributionRate = fromValueToRate(
-    //   newValue,
-    //   perSecondFeeNumerator,
-    //   perSecondFeeDenominator
-    // );
-    // const resp = await claimerContract.claim(
-    //   account,
-    //   baseCoord.toString(10),
-    //   path,
-    //   newContributionRate,
-    //   {
-    //     from: account,
-    //     value: calculateWeiSubtotalField(displayNetworkFeePayment),
-    //   }
-    // );
-    // const receipt = await resp.wait();
-    // const filter = claimerContract.filters.ParcelClaimed(null, null);
-    // const res = await claimerContract.queryFilter(
-    //   filter,
-    //   receipt.blockNumber,
-    //   receipt.blockNumber
-    // );
-    // let licenseId = res[0].args[0];
-    // return licenseId;
-    return "";
+    const baseCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase1Coord.x,
+      claimBase1Coord.y
+    );
+    const destCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase2Coord.x,
+      claimBase2Coord.y
+    );
+    let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord);
+    path.map(
+      // FIXME: it should be 'BN' type, waiting for js-geo-web-coordinate upgrade.
+      (v: any) => {
+        return BigNumber.from(v.toString(10));
+      }
+    );
+    if (path.length == 0) {
+      path = [BigNumber.from(0)];
+    }
+    const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
+    const newContributionRate = fromValueToRate(
+      newValue,
+      perSecondFeeNumerator,
+      perSecondFeeDenominator
+    );
+    const resp = await claimerContract.claim(
+      account,
+      newContributionRate,
+      // baseCoord.toString(10),
+      path,
+      {
+        from: account,
+        // value: calculateWeiSubtotalField(displayNetworkFeePayment),
+      }
+    );
+    const receipt = await resp?.wait();
```

Comment:
Seems that I did something wrong here, but I can not catch it: 👀 

` Error: invalid arrayify value (argument="value", value=, code=INVALID_ARGUMENT, version=bytes/5.4.0)`

Reply:
It's hard to decide the **3rd** parameters of `claimerContract.claim()`.

Reply:
Seems that the contract needs 2 parameters, both `path` and `baseCoord`:

https://github.com/Geo-Web-Project/core-contracts/blob/bae960170c9a7d05c5fe3394d81cc1e5967ca661/contracts/FairLaunchClaimer.sol#L52

But how to pass it?

Reply:
Found typo: `unit64` should be `uint64` when using `abiCoder.encode()`.

Will try it later.